packages/js/src/plugins/auctionHouseModule/operations/executeSale.ts

```diff
@@ -308,11 +320,15 @@ export const executeSaleBuilder = (
   if (hasAuctioneer && !auctioneerAuthority) {
     throw new AuctioneerAuthorityRequiredError();
   }
+  if (isPartialSale) {
+    const buyerPricePerToken = buyerPrice.basisPoints.div(
+      buyerTokensSize.basisPoints
+    );
 
-  // Data.
-  const price = isNative
-    ? lamports(bid.price.basisPoints)
-    : amount(bid.price.basisPoints, treasuryMint.currency);
+    if (!price.basisPoints.eq(buyerPricePerToken.mul(tokens.basisPoints))) {
```

Comment:
@lorisleiva This check copies the validation for the program side but adds more context to the user.

I don't have an idea how this should work from the Auctioneer side, since the listing price is the **u64::MAX** which requires the user to give **(u64::MAX / token size)** SOLs for each token. The same validation is in **auctioneer_execute_logic** and for me, it doesn't look right.

Reply:
Noticed in docs that partial sale isn't supported in Auctioneer

Reply:
https://docs.metaplex.com/programs/auction-house/auctioneer